sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DataFeedSourcesTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.MetricsAdvisor.Administration;
+using Azure.AI.MetricsAdvisor.Models;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.MetricsAdvisor.Tests
+{
+    public class DataFeedSourcesTests : MockClientTestBase
+    {
+        private static object[] DataSourceTestCases =
+        {
+            new object[] { new AzureApplicationInsightsDataFeedSource("mock", "secret", "mock", "mock"), "\"apiKey\":\"secret\"" },
+            new object[] { new AzureBlobDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new AzureCosmosDbDataFeedSource("secret", "mock", "mock", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new AzureDataExplorerDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new AzureDataLakeStorageGen2DataFeedSource("mock", "secret", "mock", "mock", "mock"), "\"accountKey\":\"secret\"" },
+            new object[] { new AzureTableDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new InfluxDbDataFeedSource("secret", "mock", "mock", "mock", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new InfluxDbDataFeedSource("mock", "mock", "mock", "secret", "mock"), "\"password\":\"secret\"" },
+            new object[] { new MongoDbDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new MySqlDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new PostgreSqlDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
+            new object[] { new SqlServerDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" }
+        };
+
+        public DataFeedSourcesTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private string DataFeedResponseContent => @"
+        {
+            ""dataSourceType"": ""SqlServer"",
+            ""dataSourceParameter"": {},
+            ""metrics"": []
+        }
+        ";
+
+        [Test]
+        [TestCaseSource(nameof(DataSourceTestCases))]
+        public async Task DataFeedSourceSendsSecretDuringCreation(DataFeedSource dataSource, string expectedSubstring)
```

Comment:
The service does not return the secrets anymore, so we can't check we sent them correctly in our live tests. Created a non-live test for this purpose.